File: puzzles/go/onboarding/onboarding.go
package main

import "fmt"

func main() {
  // game loop
  for {
    var enemy1 string  // name of enemy 1
    fmt.Scan(&enemy1)
    var dist1 int // distance to enemy 1
    fmt.Scan(&dist1)
    var enemy2 string // name of enemy 2
    fmt.Scan(&enemy2)
    var dist2 int // distance to enemy 2
    fmt.Scan(&dist2)
    
    // Display enemy1's name when enemy1 is the closest, enemy2's name otherwise
    if dist1 < dist2 {
        fmt.Println(enemy1)
    } else {
        fmt.Println(enemy2)
    }
  }
}

File: puzzles/cpp/onboarding/onboarding.cpp
#include <iostream>
#include <string>

using namespace std;

int main() {
  // game loop
  while (true) {
    string enemy1; // name of enemy 1
    cin >> enemy1; cin.ignore();

    // validate input
    if (enemy1.empty()) {
      return 0;
    }

    int dist1; // distance to enemy 1
    cin >> dist1; cin.ignore();
    string enemy2; // name of enemy 2
    cin >> enemy2; cin.ignore();
    int dist2; // distance to enemy 2
    cin >> dist2; cin.ignore();

    if (dist1 < dist2) {
      cout << enemy1 << endl;
    }  else {
      cout << enemy2 << endl;
    }
  }
}
